Model/Service/Index.php

```diff
@@ -62,7 +62,7 @@
 use Nosto\Tagging\Model\ResourceModel\Product\Index\CollectionFactory as NostoIndexCollectionFactory;
 use Nosto\Tagging\Util\Serializer\ProductSerializer;
 use Nosto\Tagging\Util\Iterator;
-use Nosto\Tagging\Util\Product as ProductUtil;
+use Nosto\Tagging\Util\Comparator\ProductComparatorInterface;
```

Comment:
Utility packages shouldn't have interfaces. This should be under `Model/Product/Comparator` or something. We can later on restructure the models to be services.